src/CsvBulkLoader/SaveListener.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+namespace SilverStripe\Snapshots\CsvBulkLoader;
+
+use SilverStripe\EventDispatcher\Dispatch\Dispatcher;
+use SilverStripe\EventDispatcher\Symfony\Event;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+
+class SaveListener extends DataExtension
+{
+    public function onAfterProcessRecord(DataObject $obj, $preview, $isChanged)
+    {
+        // No need tracking previews, since we don't expect any writes
+        if ($preview) {
+            return;
+        }
+
+        // Only record snapshot on changes.
+        // All rows from a CSV import will come through this by default.
+        if (!$isChanged) {
+            return;
+        }
+
+        Dispatcher::singleton()->trigger(
+            'csvBulkLoaderImport',
+            Event::create('import', ['obj' => $obj])
```

Comment:
This `Event` object is a bit weird, with this `action` parameter as you probably discovered. The idea here is that you can do filtering at the configuration level by specifying the "action" as a certain context you want to bind to. It makes sense with things like "graphqlMutation.updateFile" or "cmsAction.reorderTree", but for things like this it's not so clear.

What I'd probably do here is pass `get_class($obj)` as the action. That way you could conceivably do `on: 'csvBulkLoaderImport.SilverStripe\CMS\Model\SiteTree` to only bind to page imports, for instance.

Let's also call it "record" for consistency with other event handlers.

```suggestion
            Event::create(get_class($obj), ['record' => $obj])
```